### ISSUE TITLE
New-TestResources.ps1 should set AZURE_AUTHORITY_HOST environment variable

### ISSUE THREAD

**Issue by danieljurek** [Azure/azure-sdk-tools#757]
New-TestResources.ps1 should be able to set AZURE_AUTHORITY_HOST (https://azure.github.io/azure-sdk/general_azurecore.html#environment-variables) possibly based on the specified [$Environment](https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/TestResources/New-TestResources.ps1#L56)

**Comment by pakrym** [comment]
We might also want to provide management URIs:
```
          SERVICE_MANAGEMENT_URL: https://management.core.usgovcloudapi.net/
          RESOURCE_MANAGER: https://management.usgovcloudapi.net/
```

**Comment by danieljurek** [comment]
@pakrym - Where do those environment variable names come from? I'm using variables defined in the guidelines https://azure.github.io/azure-sdk/general_azurecore.html#environment-variables 

A cursory glance through the .NET codebase doesn't show `SERVICE_MANAGEMENT_URL` or `RESOURCE_MANAGER` used anywhere.

**Comment by pakrym** [comment]
We don't currently use these variables. This is something that came up in https://github.com/Azure/azure-sdk-for-net/pull/13217 where we need to have the management endpoint to be able to create resources as part of the test. 

I just thought that this might be a common occurrence in the future and it would be nice to always set these so all tests get access to them. We do this with other "universal"  variables:

https://github.com/Azure/azure-sdk-tools/blob/ad84f0b0bc45521bb2b75ce8c3b1aa7ae6f24e44/eng/common/TestResources/New-TestResources.ps1#L294-L302

**Comment by danieljurek** [comment]
@jsquire and I spoke today and he was able to re-enforce the necessity of that idea and where the `SERVICE_MANAGEMENT_URL` and `RESOURCE_MANAGER_URL` come from. It makes sense to put those in there and tests that use them can take advantage of those today. 

The variable names are known and the standard for converting should be `ResourceManagerUrl` -> `RESOURCE_MANAGER_URL`. For now we'll manually add the ones we need. Eventually we could start making use of things like `StorageEndpointSuffix` automatically and that could offer an option that doesn't involve generating URLs and connection strings in ARM templates. 

Environment parameters are: 

```
Name                                              : AzureCloud
EnableAdfsAuthentication                          : False
OnPremise                                         : False
ActiveDirectoryServiceEndpointResourceId          : https://management.core.windows.net/
AdTenant                                          : Common
GalleryUrl                                        : https://gallery.azure.com/
ManagementPortalUrl                               : https://go.microsoft.com/fwlink/?LinkId=254433
ServiceManagementUrl                              : https://management.core.windows.net/
PublishSettingsFileUrl                            : https://go.microsoft.com/fwlink/?LinkID=301775
ResourceManagerUrl                                : https://management.azure.com/
SqlDatabaseDnsSuffix                              : .database.windows.net
StorageEndpointSuffix                             : core.windows.net
ActiveDirectoryAuthority                          : https://login.microsoftonline.com/
GraphUrl                                          : https://graph.windows.net/
GraphEndpointResourceId                           : https://graph.windows.net/
TrafficManagerDnsSuffix                           : trafficmanager.net
AzureKeyVaultDnsSuffix                            : vault.azure.net
DataLakeEndpointResourceId                        : https://datalake.azure.net/
AzureDataLakeStoreFileSystemEndpointSuffix        : azuredatalakestore.net
AzureDataLakeAnalyticsCatalogAndJobEndpointSuffix : azuredatalakeanalytics.net
AzureKeyVaultServiceEndpointResourceId            : https://vault.azure.net
AzureOperationalInsightsEndpointResourceId        : https://api.loganalytics.io
AzureOperationalInsightsEndpoint                  : https://api.loganalytics.io/v1
AzureAnalysisServicesEndpointSuffix               : asazure.windows.net
AnalysisServicesEndpointResourceId                : https://region.asazure.windows.net
AzureAttestationServiceEndpointSuffix             : attest.azure.net
AzureAttestationServiceEndpointResourceId         : https://attest.azure.net
```

**Comment by jsquire** [comment]
I believe this was addressed by [#822](https://github.com/Azure/azure-sdk-tools/pull/822), but I'll defer to @danieljurek to make the call and, if so, close this out.

**Comment by jsquire** [comment]
With the closing of #828, I believe the work tracked by this item is now complete.   Closing this out,   @danieljurek:  Please feel free to reopen if I've overlooked something.
